starter/src/main/java/com/starter/fullstack/dao/InventoryDAO.java

```diff
@@ -82,7 +83,11 @@ public Optional<Inventory> update(String id, Inventory inventory) {
    * @return Deleted Inventory.
    */
   public Optional<Inventory> delete(String id) {
-    // TODO
-    return Optional.empty();
+    Inventory deleted = this.mongoTemplate.findById(id, Inventory.class);
```

Comment:
This does work, but I'd challenge you to look through mongoTemplate's documentation to see if you can find a way to find and delete with a single database call.

You always want to try to make the least amount of database calls to maximize performance